src/main/java/com/monitorjbl/plugins/UserUtils.java

```diff
@@ -19,44 +23,44 @@ public UserUtils(UserService userService) {
     this.userService = userService;
   }
 
-  public List<User> dereferenceUsers(Iterable<String> users) {
-    List<User> list = newArrayList();
-    for(String u : users) {
-      Optional<User> user = getUserByName(u);
-      if(user.isPresent()) {
-        list.add(user.get());
-      }
+  public List<User> dereferenceUsers(Collection<String> usernames) {
+    Set<String> distinctNames = new HashSet<>(usernames);
+    if (distinctNames.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    Set<ApplicationUser> usersByName = userService.getUsersByName(distinctNames);
```

Comment:
Using `getUsersByName` here allows the server to perform a significantly more efficient lookup, retrieving users with a single database query instead of one per user.

Reply:
What happens if there are a lot of users? Is this paged?